src/util.c

```diff
@@ -1147,6 +1148,191 @@ int reclaimFilePageCache(int fd, size_t offset, size_t length) {
 #endif
 }
 
+/** An async signal safe version of fgets().
+ * Has the same behaviour as standard fgets():  reads a line from fd and stores it into the dest buffer.
+ * It stops when either (buff_size-1) characters are read, the newline character is read, or the end-of-file is reached,
+ *  whichever comes first. */
+char *fgets_async_signal_safe(char *dest, int buff_size, int fd) {
+    for (int i = 0; i < buff_size; i++) {
+        /* Read one byte */
+        ssize_t bytes_read_count = read(fd, dest + i, 1);
+        /* On EOF return NULL */
+        if (!bytes_read_count) {
+            return NULL;
+        }
+        /* we found the end of the line. */
+        if (dest[i] == '\n') {
+            break;
+        }
+    }
+    return dest;
+}
+
+static const char HEX[] = "0123456789abcdef";
+
+static char *safe_utoa(int _base, uint64_t val, char *buf) {
```

Comment:
actually, i copied it as is from Twitter, but you are right and that it will be better to align all the new safe functions' names.
I'd go with <orig_name>_async_signal_safe